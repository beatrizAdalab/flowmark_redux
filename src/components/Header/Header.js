import React from 'react';
import { Link } from 'react-router-dom';
import { LoginConsumer } from '../../context/LoginContext';

// fontawesome imports
import { faPowerOff, faHome } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

function Header({ user, logoutUser }) {
    return (
        <nav className='navbar navbar-light bg-white fixed-top'>
            <div className='container'>
                <h1 className='navbar-brand m-0 ml-2 '> <span className='text-info'>F</span>lowMark</h1>
                <div className='d-flex justify-content-center align-items-center'>
                    <p class="m-0 pr-2 text-info">{user.userName}</p>
                    <Link className='text-decoration-none' to='/listClassifieds/'>
                        <div className='d-flex align-items-center'>
                            <FontAwesomeIcon className='text-info' icon={faHome} />
                        </div>
                    </Link>

                    <button
                        className='btn btn-link'
                        onClick={logoutUser}>
                        <FontAwesomeIcon className='text-info' icon={faPowerOff} />
                    </button>

                </div>
            </div>
        </nav>
    )
}

export default Header;