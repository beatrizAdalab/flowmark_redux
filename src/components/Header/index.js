import Header from './Header';
import { connect } from 'react-redux';
import { getUi, getUser } from '../../store/selectors';
import { logout } from '../../store/actions';


function mapStateToProps(state, ownProps) {
  return {
    user: getUser(state)
  };
}

const mapDispatchToProps = {
  logoutUser: logout,
};

const connected = connect(mapStateToProps, mapDispatchToProps);
const HeaderConnected = connected(Header);

export default HeaderConnected;