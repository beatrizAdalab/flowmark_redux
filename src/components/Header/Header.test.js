import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';


describe('Header', () => {
  const logout = {
    login: false,
    register: false,
    errorLogin: '',
    errorRegister: '',
    userName: '',
    userPassword: ''
  };

  const props = {
    user: {
      login: true,
      register: false,
      errorLogin: '',
      errorRegister: '',
      userName: 'userTwo34',
      userPassword: '34'
    },
    logoutUser: jest.fn().mockResolvedValue(logout)
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Header {...props} />);
  });

  test('snapshot testing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render', async () => {
    expect(wrapper.exists()).toBe(true);
  });

  test('should logout to user', () => {
    wrapper.find('button').simulate('click');
    expect(props.logoutUser).toHaveBeenCalled();
  });
});