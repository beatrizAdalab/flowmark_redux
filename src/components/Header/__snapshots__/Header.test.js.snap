// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot testing 1`] = `
<nav
  className="navbar navbar-light bg-white fixed-top"
>
  <div
    className="container"
  >
    <h1
      className="navbar-brand m-0 ml-2 "
    >
       
      <span
        className="text-info"
      >
        F
      </span>
      lowMark
    </h1>
    <div
      className="d-flex justify-content-center align-items-center"
    >
      <p
        className="m-0 pr-2 text-info"
      >
        userTwo34
      </p>
      <Link
        className="text-decoration-none"
        to="/listClassifieds/"
      >
        <div
          className="d-flex align-items-center"
        >
          <FontAwesomeIcon
            border={false}
            className="text-info"
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  576,
                  512,
                  Array [],
                  "f015",
                  "M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z",
                ],
                "iconName": "home",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
        </div>
      </Link>
      <button
        className="btn btn-link"
        onClick={[MockFunction]}
      >
        <FontAwesomeIcon
          border={false}
          className="text-info"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f011",
                "M400 54.1c63 45 104 118.6 104 201.9 0 136.8-110.8 247.7-247.5 248C120 504.3 8.2 393 8 256.4 7.9 173.1 48.9 99.3 111.8 54.2c11.7-8.3 28-4.8 35 7.7L162.6 90c5.9 10.5 3.1 23.8-6.6 31-41.5 30.8-68 79.6-68 134.9-.1 92.3 74.5 168.1 168 168.1 91.6 0 168.6-74.2 168-169.1-.3-51.8-24.7-101.8-68.1-134-9.7-7.2-12.4-20.5-6.5-30.9l15.8-28.1c7-12.4 23.2-16.1 34.8-7.8zM296 264V24c0-13.3-10.7-24-24-24h-32c-13.3 0-24 10.7-24 24v240c0 13.3 10.7 24 24 24h32c13.3 0 24-10.7 24-24z",
              ],
              "iconName": "power-off",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
    </div>
  </div>
</nav>
`;
